director/director.go

```diff
@@ -1156,15 +1156,33 @@ func getHealthTestFile(ctx *gin.Context) {
 	}
 }
 
+func collectClientVersionMiddleware(c *gin.Context) {
+	userAgentSlc := c.Request.Header["User-Agent"]
+	if len(userAgentSlc) < 1 {
+		log.Error("Failed to parse User-Agent")
+	}
+
+	userAgent := userAgentSlc[0]
+
+	uaRegExp := regexp.MustCompile(`^pelican-[^\/]+\/\d+\.\d+\.\d+`)
+	if matches := uaRegExp.MatchString(userAgent); !matches {
+		log.Error("Failed to match User-Agent for Pelican client version")
```

Comment:
Since we use this same regex in other parts of the Director, let's pull it out and functionalize it to guarantee that we can't update one but forget the other.

A second note - I don't know if this really constitutes an "error" because we often use non-pelican clients to interact with the federation as well, such as `curl`. After reading the Cloudflare article you sent, I think we want to avoid creating any prom labels based on an arbitrary incoming UA header, but we should think about how to make this flexible enough to handle the case where the incoming UA isn't Pelican. Maybe it's enough to have a single label that tracks non-pelican clients?

Lastly, w.r.t. cardinality explosion and Pelican headers, do you have a sense of the "cost" of tracking the 56 versions we've already released, knowing that we release between 3-5 new versions every month?